pkg/shared/logging.go

```diff
@@ -47,11 +52,22 @@ func LogInfo(
 		cr.Namespace,
 		cr.Name,
 	)
+
+	if logEvent {
+		LogEvent(
+			cr,
+			v1.EventTypeNormal,
+			eventReason,
+			msg,
+		)
+	}
 }
 
 // LogInfof logs the given message format and payload at Info level.
 func LogInfof(
 	cr *kdv1.KubeDirectorCluster,
+	logEvent bool,
+	eventReason string,
```

Comment:
Seems a bit wonky to have two inputs here, since there are "illegal" combos (true + empystring, or false + string).

I think we should just have one argument, the reason, and if it is emptystring then don't make an event. Or we could even make it stringpointer so that we can use nil as the "no event" value? Not sure.

Reply:
pointer makes sense.. will do

Reply:
👍